third_party/github.com/hashicorp/terraform-provider-google-beta/google-beta/services/vertexai/resource_vertex_ai_endpoint.go

```diff
@@ -1004,5 +1010,91 @@ func expandVertexAIEndpointEncryptionSpecKmsKeyName(v interface{}, d tpgresource
 }
 
 func expandVertexAIEndpointNetwork(v interface{}, d tpgresource.TerraformResourceData, config *transport_tpg.Config) (interface{}, error) {
-	return v, nil
+	// This API uses a unique formatting of network different from the standard self_link URI.
+	// We need to replace the project_id with project_number when expanding the value of network field.
+	return transformProjectIDToProjectNumberInNetwork(v, d, config)
+}
+
+func transformProjectIDToProjectNumberInNetwork(v interface{}, d tpgresource.TerraformResourceData, config *transport_tpg.Config) (interface{}, error) {
+	if v == nil {
+		return v, nil
+	}
+	raw, ok := v.(string)
+	if !ok {
+		return nil, fmt.Errorf("unexpected type %T", v)
+	}
+	if raw == "" {
+		return raw, nil
+	}
+	projectID, networkName, err := parseNetwork(raw)
+	if err != nil {
+		return nil, err
+	}
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return nil, err
+	}
+	projectNumber, err := getProjectNumber(config, projectID, userAgent)
+	if err != nil {
+		return nil, err
+	}
+	transformed := fmt.Sprintf("projects/%s/global/networks/%s", projectNumber, networkName)
+	return transformed, nil
+}
+
+func transformProjectNumberToProjectIDInNetwork(v interface{}, d tpgresource.TerraformResourceData, config *transport_tpg.Config) (interface{}, error) {
+	if v == nil {
+		return v, nil
+	}
+	raw, ok := v.(string)
+	if !ok {
+		return nil, fmt.Errorf("unexpected type %T", v)
+	}
+	if raw == "" {
+		return raw, nil
+	}
+	projectNumber, networkName, err := parseNetwork(raw)
+	if err != nil {
+		return nil, err
+	}
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return nil, err
+	}
+	projectID, err := getProjectID(config, projectNumber, userAgent)
+	if err != nil {
+		return nil, err
+	}
+	transformed := fmt.Sprintf("projects/%s/global/networks/%s", projectID, networkName)
+	return transformed, nil
+}
+
+func getProjectNumber(config *transport_tpg.Config, project, userAgent string) (string, error) {
+	getProjectCall := config.NewResourceManagerClient(userAgent).Projects.Get(project)
```

Comment:
Do you know if it is possible to reference a network in a different project? I'm curious if we should add warnings about permissions. (Though we can always improve the doc/workaround the issue incrementally.)

Reply:
In theory it is possible as long as the GSA has the `resourcemanager.project.get` permission. If users gets confused by it, we could add a warning about permission later.